api/app/signals/apps/signals/tasks.py

```diff
@@ -86,3 +87,22 @@ def anonymize_reporter(reporter_id):
                 'text': text,
                 'created_by': None  # This wil show as "SIA systeem"
             }, signal=reporter.signal)
+
+
+@app.task
+def update_status_children_based_on_parent(signal_id):
+    if not settings.FEATURE_FLAGS.get('TASK_UPDATE_CHILDREN_BASED_ON_PARENT', True):
+        # Feature disabled
+        log.warning("Feature 'TASK_UPDATE_CHILDREN_BASED_ON_PARENT' disabled!")
+        return
+
+    text = 'Hoofdmelding is afgehandeld hierdoor is deze deelmelding geannuleerd'
+    signal = Signal.objects.get(pk=signal_id)
+    if signal.is_parent() and signal.status.state in [AFGEHANDELD, ]:
+        # Lets check the children
+        children = signal.children.exclude(status__state__in=[AFGEHANDELD, GEANNULEERD, ])
+        for child in children:
+            # All children must get the state "GEANNULEERD"
+            status = Status.objects.create(state=GEANNULEERD, text=text, _signal=child)
+            child.status = status
+            child.save()
```

Comment:
The choice not to go through the `SignalManager` or fire an appropriate Django signal means that these status changes cannot be picked up by other parts of SIA. For example a search index would not be updated --- although right now we do not track the status field. I am aware that dealing with Django Signals leading to more Django Signals may be tricky.

Consider perhaps adding a `cancel_children` action to the SignalManager and generating appropriate Django Signals. (Still called from this task, so that it does not happen in the request-response cycle.)

Reply:
No need for adding a "new" action to the SignalsManager. We can just use the existing code. Will change it to use that.

Reply:
Done